spec/jobs/trs_update_mentor_details_job_spec.rb

```diff
@@ -0,0 +1,115 @@
+require "rails_helper"
+
+RSpec.describe TRSUpdateMentorDetailsJob, type: :job do
+  describe "#perform" do
+    context "when the mentor has a valid trn" do
+      let(:mentor) { create(:mentor, first_name: "Joe", last_name: "Bloggs", trn: "1234567") }
```

Comment:
Same as above, could add a bang (`let!(:mentor)`) and then you wouldn't need to call it in the before block